crates/cheatcodes/src/evm.rs

```diff
@@ -567,6 +567,14 @@ impl Cheatcode for stopAndReturnStateDiffCall {
     }
 }
 
+
+impl Cheatcode for setBlockhashCall{
+    fn apply_full<DB: DatabaseExt>(&self, ccx: &mut CheatsCtxt<DB>) -> Result {
+        todo!()
```

Comment:
here we need a way to insert the blockhash in the db.
this isn't exposed via the regular `Database` trait, so we need to add a function to the `DatabaseExt` trait

https://github.com/foundry-rs/foundry/blob/32f01e3003bc4a98691282c5a03661214e3f5645/crates/evm/core/src/backend/mod.rs#L71-L71
and implement it manually:

https://github.com/foundry-rs/foundry/blob/32f01e3003bc4a98691282c5a03661214e3f5645/crates/evm/core/src/backend/mod.rs#L878-L878